src/dashboard/src/devine_dashboard/js/dashboard/canvas_drawer.js

```diff
@@ -0,0 +1,319 @@
+   */
+  _setColor(color) {
+    if (color) {
+      this.canvas.strokeStyle = color;
+      this.canvas.fillStyle = color;
+    }
+  }
+
+  /** Draw an old-school style ne camera feed warning */
+  _drawNoFeed() {
+    this._setColor("red");
+    this.canvas.font = "bold 20pt Arial";
+    this.canvas.fillText("< No Camera Feed />", 190, this.canvasSize.y / 2);
+  }
+
+  /**
+   * Draw a cross an the position found
+   * @param {int} x - X coordinate.
+   * @param {int} y - Y coordinate.
+   */
+  _drawPositionFound(x, y) {
+    this._setColor("#3498DB");
+    this.canvas.fillText("Found!", x + 8, y - 8);
+    this.canvas.fillRect(x - 3, y - 3, 6, 6);
+    this.canvas.fillRect(x - 20, y - 1, 40, 2);
+    this.canvas.fillRect(x - 1, y - 20, 2, 40);
+  }
+
+  /**
+   * Draw rectangle over object found on the image.
+   * @param {array} objects - The list of objects.
+   * @param {int|undefined} confidence - The confidence value, if an object was found.
+   */
+  _drawObjectsRectangles(objects, confidence) {
+    for (var i = 0; i < objects.length; i++) {
+      this._setColor(distinctColors[i % 14]);
+
+      this.canvas.beginPath();
+      let {x_offset, y_offset, height, width} = objects[i].bounding_box;
+
+      this.canvas.rect(x_offset, y_offset, width, height);
+
+      this.canvas.fillText(objects[i].category_name, x_offset, y_offset - 1);
+
+      // Show the confidence about the object found, if any.
+      if (confidence !== undefined && confidence.length === objects.length) {
+        let obj_name_offset = this.canvas.measureText(objects[i].category_name).width;
+        this.canvas.fillText(`(${(confidence[i]*100).toFixed(2)}%)`, x_offset + obj_name_offset + 5, y_offset - 1);
+      }
+
+      this.canvas.stroke();
+      this.canvas.closePath();
+    }
+  }
+  
+  /**
+   * Clear the canvas and reset the image.
+   * @param {object} imageObject - The image data.
+   */
+  _resetCanvas(imageObject) {
+    this.canvas.clearRect(0, 0, this.canvasSize.x, this.canvasSize.y);
+    this.canvas.drawImage(imageObject, 0, 0);
+    this.canvas.font = "bold 12pt Arial";
+    this.canvas.lineWidth = "2";
+  }
+  
+  /**
+   * Retreive the current history element with the current position.
+   * @param {array} historyArray - The array to take the element from.
+   * @return {object} The element.
+   */
+  _getCurrentHistoryElement(historyArray, stamp=null) {
+    if (!stamp) {
+      // Return latest element if there is no stamp
+      return historyArray[historyArray.length - this.history.position];
+    }
+    for (let i = historyArray.length - 1; i >= 0; --i) {
+      if (Math.abs(historyArray[i].header.stamp.secs - stamp.secs) < 2) {
+        return historyArray[i];
+      }
+    }
+  }
+
+  /**
+   * Map body part with one another (e.g.: ear with eye).
+   * Adaptation and refactor of function draw_humans in tf_pose/estimator.py
+   * @param {array} humans - The list of humans detected.
+   *
+   * @see CocoPairs in tf_pose/common.py
```

Comment:
The comment about cocopairs should maybe stay with the `BODY_PARTS` constant. Up to you

Reply:
Good point